src/main/java/pl/betoncraft/betonquest/objectives/EntityInteractObjective.java

```diff
@@ -0,0 +1,248 @@
+		if (name != null) {
+			name = name.replace('_', ' ');
+		}
+		marked = instruction.getOptional("marked");
+		if (marked != null) {
+			marked = Utils.addPackage(instruction.getPackage(), marked);
+		}
+		notify = instruction.hasArgument("notify");
+	}
+
+	@Override
+	public void start() {
+		switch (interaction) {
+		case RIGHT_CLICK:
+			rightClickListener = new RightClickListener();
+			break;
+		case DAMAGE:
+			damageListener = new DamageListener();
+			break;
+		case ANY:
+			rightClickListener = new RightClickListener();
+			damageListener = new DamageListener();
+			break;
+		}
+	}
+
+	private void onInteract(Player player, Entity entity) {
+		// check if it's the right entity type
+		if (!entity.getType().equals(mobType)) {
+			return;
+		}
+		// if the entity should have a name and it does not match, return
+		if (name != null && (entity.getCustomName() == null || !entity.getCustomName().equals(name))) {
+			return;
+		}
+		// check if the entity is correctly marked
+		if (marked != null) {
+			if (!entity.hasMetadata("betonquest-marked")) {
+				return;
+			}
+			List<MetadataValue> meta = entity.getMetadata("betonquest-marked");
+			for (MetadataValue m : meta) {
+				if (!m.asString().equals(marked)) {
+					return;
+				}
+			}
+		}
+		// check if the player has this objective
+		String playerID = PlayerConverter.getID(player);
+		if (containsPlayer(playerID) && checkConditions(playerID)) {
+			// get data off the player
+			EntityInteractData playerData = (EntityInteractData) dataMap.get(playerID);
+			// check if player already interacted with entity
+			if (playerData.containsEntity(entity))
+				return;
+			// right mob is interacted with, handle data update
+			playerData.subtract();
+			playerData.addEntity(entity);
+			if (playerData.isZero()) {
+				completeObjective(playerID);
+			} else if (notify) {
+				// send a notification
+				Config.sendMessage(playerID, "mobs_to_kill", new String[] { String.valueOf(playerData.getAmount()) });
+			}
+		}
+	}
+
+	@Override
+	public void stop() {
+		if (rightClickListener != null)
+			HandlerList.unregisterAll(rightClickListener);
+		if (damageListener != null)
+			HandlerList.unregisterAll(damageListener);
+	}
+
+	@Override
+	public String getDefaultDataInstruction() {
+		return Integer.toString(amount);
+	}
+
+	@Override
+	public String getProperty(String name, String playerID) {
+		if (name.equalsIgnoreCase("left")) {
+			return Integer.toString(((EntityInteractData) dataMap.get(playerID)).getAmount());
+		} else if (name.equalsIgnoreCase("amount")) {
+			return Integer.toString(amount - ((EntityInteractData) dataMap.get(playerID)).getAmount());
+		}
+		return "";
+	}
+
+	private static class EntityInteractData extends ObjectiveData {
+
+		private int amount;
+		private Set<UUID> entitys;
+
+		public EntityInteractData(String instruction, String playerID, String objID) {
+			super(instruction, playerID, objID);
+			String[] args = instruction.split("-");
```

Comment:
Here's a problem. UUIDs look like this: `718a5bd7-90cf-4e01-be6f-792d84e1b7a0`
There are `-` characters there, so splitting the instruction with them will yield a lot of cut UUIDs.

I suggest to use space as a separation of the instruction. That way `[0]` index will always be the amount left, and all other elements of that array will be the UUIDs of entities.